crates/bevy_pbr/src/atmosphere/functions.wgsl

```diff
@@ -334,9 +348,16 @@ fn ndc_to_uv(ndc: vec2<f32>) -> vec2<f32> {
 }
 
 /// Converts a direction in world space to atmosphere space
-fn direction_world_to_atmosphere(dir_ws: vec3<f32>) -> vec3<f32> {
-    let dir_as = atmosphere_transforms.atmosphere_from_world * vec4(dir_ws, 0.0);
-    return dir_as.xyz;
+fn direction_world_to_atmosphere(dir_ws: vec3<f32>, up: vec3<f32>) -> vec3<f32> {
+    // Camera forward in world space (+Z in view-to-world transform here)
+    let forward_ws = normalize((view.world_from_view * vec4(0.0, 0.0, 1.0, 0.0)).xyz);
+    let tangent_z = normalize(forward_ws - up * dot(forward_ws, up));
+    let tangent_x = normalize(cross(up, tangent_z));
+    return vec3(
+        dot(dir_ws, tangent_x),
+        dot(dir_ws, up),
+        dot(dir_ws, tangent_z),
+    );
```

Comment:
this feels like it would be simplified. the return value is a matmul, the first term shouldnt need a normalize, and should just be a `view.world_from_view[2].xyz` i believe? and the second term looks like a vector rejection (which can produce zero when forward_ws = up which normalizes to NaN) and the last one doesnt need a normalize either as the first two are already orthogonal unit length.